From 7f95f4c921c03ae5c1b252004c748d440f54c501 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lothar=20Wa=C3=9Fmann?= <LW@KARO-electronics.de>
Date: Thu, 16 Jan 2020 12:25:41 +0100
Subject: [PATCH 22/35] net: phy: micrel: support storing the skew parameters
 in the PHY's DT node(s) rather than the MAC's

Lookup 'phy-handle' to find the related phy node for looking up the
skew settings.
If there is no 'phy-handle' property the values will be read from the
MAC's DT node as before.
---
 drivers/net/phy/micrel_ksz90x1.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/net/phy/micrel_ksz90x1.c b/drivers/net/phy/micrel_ksz90x1.c
index 7a8ea3c..6b411d6 100644
--- a/drivers/net/phy/micrel_ksz90x1.c
+++ b/drivers/net/phy/micrel_ksz90x1.c
@@ -114,12 +114,21 @@ static int ksz90x1_of_config_group(struct phy_device *phydev,
 	const int ps_to_regval = 60;
 	int i, changed = 0, offset, max;
 	u16 regval = 0;
+	struct ofnode_phandle_args phandle_args;
+	ofnode of_node = dev_ofnode(dev);
+	int ret;
 
 	if (!drv || !drv->writeext)
 		return -EOPNOTSUPP;
 
+	ret = dev_read_phandle_with_args(dev, "phy-handle", NULL, 0, 0,
+					 &phandle_args);
+	if (ret == 0)
+		of_node = phandle_args.node;
+
 	for (i = 0; i < ofcfg->grpsz; i++) {
-		int val = dev_read_u32_default(dev, ofcfg->grp[i].name, ~0);
+		int val = ofnode_read_u32_default(of_node, ofcfg->grp[i].name,
+						  ~0);
 
 		offset = ofcfg->grp[i].off;
 		if (val == -1) {
-- 
2.7.4

